src/main/java/io/jcoder/tutorials/ch12/CLI.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2020 - JCoder Ltd
+ */
+package io.jcoder.tutorials.ch12;
+
+import java.util.InputMismatchException;
+import java.util.Scanner;
+
+/**
+ * Command Line Interface Class used in the Vending Machine Project.
+ * 
+ * <p>
+ * https://jcoder.io/content/course/java/beginners/ch12/
+ * </p>
+ * 
+ * @author Andres Ricardo Perez
+ */
+public class CLI {
+
+	private VendingMachine machine;
+
+	public CLI(VendingMachine machine) {
+		this.machine = machine;
+	}
+
+	public void runVendingMachine() {
```

Comment:
This is heading in the right direction :) - can you split it into submethods please?

For example:
`printBalance`
`printAvailableProducts`
`printOptions`
`readOption` ...